rotkehlchen/inquirer.py

```diff
@@ -500,6 +536,154 @@ def find_uniswap_v2_lp_price(
 
+    def find_curve_pool_price(
+        self,
+        lp_token: EthereumToken,
+    ) -> Optional[Price]:
+        """
+        1. Obtain the pool for this token
+        2. Obtain prices for assets in pool
+        3. Obtain the virtual price for share and the balances of each
+        token in the pool
+        4. Calc the price for a share
+
+        Returns the price of 1 LP token from the pool
+        """
+        assert self._ethereum is not None, 'Inquirer ethereum manager should have been initialized'  # noqa: E501
+
+        pools = get_curve_pools()
+        if lp_token.ethereum_address not in pools:
+            return None
+        pool = pools[lp_token.ethereum_address]
+        tokens = []
+        # Translate addresses to tokens
+        try:
+            for asset in pool.assets:
+                if asset == '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE':
+                    tokens.append(A_WETH)
+                else:
+                    tokens.append(EthereumToken(asset))
+        except UnknownAsset:
+            return None
+
+        # Get price for each token in the pool
+        prices = []
+        for token in tokens:
+            price = self.find_usd_price(token)
+            if price == Price(ZERO):
+                log.error(
+                    f'Could not calculate price for {lp_token} due to inability to '
+                    f'fetch price for {token}.',
+                )
+                return None
+            prices.append(price)
+
+        # Query virtual price of LP share and balances in the pool for each token
+        contract = EthereumContract(
+            address=pool.pool_address,
+            abi=CURVE_POOL_ABI,
+            deployed_block=0,
+        )
+        calls = [(pool.pool_address, contract.encode(method_name='get_virtual_price'))]
+        calls += [
+            (pool.pool_address, contract.encode(method_name='balances', arguments=[i]))
+            for i in range(len(pool.assets))
+        ]
+        output = multicall_2(
+            ethereum=self._ethereum,
+            require_success=False,
+            calls=calls,
+        )
+
+        # Check that the output has the correct structure
+        if not all([len(call_result) == 2 for call_result in output]):
+            log.debug(
+                f'Failed to query contract methods while finding curve pool price. '
+                f'Not every outcome has length 2. {output}',
+            )
+            return None
+        # Check that all the request were successful
+        if not all([contract_output[0] for contract_output in output]):
+            log.debug(f'Failed to query contract methods while finding curve price. {output}')
+            return None
+        # Deserialice information obtained in the multicall execution
+        data = []
+        virtual_price_decoded = contract.decode(output[0][1], 'get_virtual_price')
+        # Verify that the output from the contrat call is correct
```

Comment:
```suggestion
        # Verify that the output from the contract call is correct
```

But better yet just remove this. The function name is sufficient.